### PR TITLE
[Merged by Bors] - Don't use the UIBundle's Transform Fields

### PR DESCRIPTION
**Pull request by Carter0**
# Objective

I was working with the TextBundle component bundle because I wanted to change the position of the text that the bundle was holding. I used the transform field on the TextBundle at first because that is normally what controls the position of sprites in Bevy and that's what I was used to working with. 

But the actual way to change the position of text inside of a TextBundle is to use the Style's position field, not the TextBundle's transform field. 

Anecdotally, it was mentioned on the discord that other users have had this issue too. 

## Solution

I added a small doc comment to the TextBundle's transform telling users not to use it to set the position of text. And since this issue applies to the other UI bundles, I added comments there as well!

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
This is a great start! IMO we should have similar warnings on the `GlobalTransform` field, and replicate this across the other UI bundles (`NodeBundle`, `ImageBundle`, `ButtonBundle`).

File: crates/bevy_ui/src/entity.rs [hunk 1]
```diff
@@ -80,6 +80,8 @@ pub struct TextBundle {
     /// Whether this node should block interaction with lower nodes
     pub focus_policy: FocusPolicy,
     /// The transform of the node
+    /// This field is not to be used to set the position of the text inside the TextBundle.
+    /// Use the Style's position field to set text position.
```
Comment (TimJentzsch) [inline crates/bevy_ui/src/entity.rs:84]: Small formatting nit:
```suggestion
    /// Use the [`Style`]'s `position` field to set text position.
```

**Comment by DJMcNab** [comment]
Can we use nested bundles here, somehow. E.g. `SetBySystems<TransformBundle>`

**Comment by alice-i-cecile** [comment]
> SetBySystems<TransformBundle>

I'm not sure what you mean here?

I do think that documentation and code deduplication is a compelling argument for nested bundles here; I ran into similar issues when splitting up the `Style` component. @cart was opposed as a matter of style the last time we asked though.

**Comment by mockersf** [comment]
Without going into design discussion, just adding to all UI bundles that the `Transform` fields are not supposed to be manually changed could be a good start 👍

**Review by TimJentzsch — APPROVED** [review]

**Comment by DJMcNab** [comment]
`SetBySystems` would be a unit documentation bundle that the inner bundle's items should not be modified at creation time, as they'll be *set by systems*, hence the name.

**Comment by rparrett** [comment]
This seems like a good idea, although the specific language "not to be used" sounds a bit harsh to me and it seems like instructing users to use `style.position` could be a bit misleading with regard to how positioning of UI elements is typically done in Bevy.

I'd suggest something like

> This field is automatically managed by the UI layout system. To alter the position of the `NodeBundle`, use the properties of the `Style` component.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by rparrett** [comment]
This is how things appear rendered:
image

An extra empty line and/or period after "The transform of the node" would help.

Maybe bring back the link to `Style` too?

**Comment by Carter0** [comment]
> This is how things appear rendered: image
> 
> An extra empty line and/or period after "The transform of the node" would help.
> 
> Maybe bring back the link to `Style` too?

Yeah I just fixed that haha, my b. I also added the periods. 

And yeah, I can bring back the link, good point!

**Review by alice-i-cecile — APPROVED** [review]

**Comment by rparrett** [comment]
I'm not seeing the line break and/or periods, so the rendered output is still turning into a big one-line run-in sentence.

```
    /// The transform of the node.
    ///
    /// This field is automatically managed by the UI layout system.
```

`NodeBundle` should also probably get its backticks back (but no need to link that)

**Comment by Carter0** [comment]
Should be all good now. Thanks for the feedback! 

I don't know why you are not seeing the periods tho, they should be added.

**Review by rparrett — APPROVED** [review]

**Comment by rparrett** [comment]
> Oh unless you mean after node? I left that alone because none of the other comments I didn't write have periods on them.

Looks good to me. Don't need a period if there's a line break.

**Review by Carter0 — COMMENTED** [review]
Thanks for the comments everyone!

**Pull request closed without merging**